orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -91,22 +90,19 @@ public void verifyInstanceTermination(String serverGroupNameFromStage,
     return Optional.ofNullable((String) instance.orElse(new HashMap<>()).get("serverGroup"));
   }
 
-  public void verifyTrafficRemoval(String serverGroupName, String account, Location location, String cloudProvider, String operationDescriptor) {
-    Names names = Names.parseName(serverGroupName);
-
-    if (!hasDisableLock(names.getCluster(), account, location)) {
+  public void verifyTrafficRemoval(String serverGroupName, Moniker serverGroupMoniker, String account, Location location, String cloudProvider, String operationDescriptor) {
+    if (!hasDisableLock(serverGroupMoniker, account, location)) {
       return;
     }
 
-    verifyOtherServerGroupsAreTakingTraffic(serverGroupName, location, account, cloudProvider, operationDescriptor);
+    verifyOtherServerGroupsAreTakingTraffic(serverGroupName, serverGroupMoniker,location, account, cloudProvider, operationDescriptor);
   }
 
-  private void verifyOtherServerGroupsAreTakingTraffic(String serverGroupName, Location location, String account, String cloudProvider, String operationDescriptor) {
-    Names names = Names.parseName(serverGroupName);
-    Optional<Map> cluster = oortHelper.getCluster(names.getApp(), account, names.getCluster(), cloudProvider);
+  private void verifyOtherServerGroupsAreTakingTraffic(String serverGroupName, Moniker serverGroupMoniker, Location location, String account, String cloudProvider, String operationDescriptor) {
```

Comment:
Do we still need to pass around `serverGroupName` if we've got a moniker? Maybe we do now as a transition but couldn't you compare moniker's instead of `!serverGroupName.equals(tsg.getName()) &&`?

Reply:
The server group name isn't contained in th moniker object (only app, stack, detail, cluster, sequence). In the case of the Frigga naming convention, you can reconstruct the name using the moniker object. But that reconstruction won't work for the manifest based k8s provider.

Reply:
Is it silly to ask why we couldn’t have it also contain the serverGroupName?

Reply:
@lwander can probably give you a more thorough explanation. From my POV, I think that it isn't necessary to facilitate the migration away from frigga. It would also require a significant amount of effort. If you search for `serverGroupName` in Orca there are 362 occurrences.

FYI: This is related to another point. There is now a lot of redundancy with application/stack/detail in the stage data payload. In many places, those fields are already included at the root level of the stage and are now in the moniker as well. A follow up task will probably be to consolidate those.

Reply:
No that's a good question. We don't want the moniker to be the identifier for the object since the point of the moniker is to decouple the name from the relationships (app, cluster) that frigga presented us. The separation of concerns is: `name` is a way to identify an object and `moniker` is to decide which cluster & application an object belongs to. 

When you assign or derive a moniker from an object, that object doesn't even have to provide us with a name, (e.g. ec2 tags), so the idea is really to keep these concepts separate.

Reply:
My point about including the name was somewhat predicated in that when I hear `Moniker`, I think of name. 

The relationships are really part of the identity of an object, which is why we need to pass application/cluster/region/name/etc. around everywhere.

My biggest grief is that we're slowly developing more and more ways to specify the coordinates for an object: serverGroupName, asgName, regions, region, zone, zones, moniker, etc. 

In the case of moniker it's only dealing with names, so we can ignore regions/region/zone/zones and think about it only as a partial coordinate.

I hold out hope that there will be some consolidation, as it's difficult and error prone to keep everything straight.

Just my 2c. 

I think it would be cleaner and possible to pass a Moniker into validateClusterStatus but otherwise this PR fits the existing patterns and can be merged.

Reply:
My hope is this kind of clean up/refactor will point out more places where we can consolidate the many names/coordinates you point out. We'll keep this in mind going forward.